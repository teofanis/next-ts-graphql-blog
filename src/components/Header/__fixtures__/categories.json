{
  "categories": [
    {
      "name": "React",
      "slug": "react"
    },
    {
      "name": "Redux",
      "slug": "redux"
    },
    {
      "name": "React Native",
      "slug": "react-native"
    }
  ]
}
