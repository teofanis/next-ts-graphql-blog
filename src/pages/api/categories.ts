import { getCategories } from '@services/index'
import type { NextApiRequest, NextApiResponse } from 'next'

export default async function categories(
  req: NextApiRequest,
  res: NextApiResponse
) {
  try {
    const result = getCategories()
    return res.status(200).send(result)
  } catch (error) {
    return res.status(500).send(error)
  }
}
