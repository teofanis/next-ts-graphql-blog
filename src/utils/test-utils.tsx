import { render, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react'
// Add in any providers here if necessary:

const Providers: React.FC = ({ children }) => {
  return <>{children}</>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
// override render method
export { customRender as render }
