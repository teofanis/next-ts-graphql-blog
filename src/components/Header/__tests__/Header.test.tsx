import { Header } from '@components/index'
import * as Services from '@services/index'
import { act, render, screen } from '@utils/test-utils'
import React from 'react'
import { categories } from '../__fixtures__/categories.json'

const getCategoriesMock = jest.fn(() => Promise.resolve(categories))
const ServiceSpy = jest.spyOn(Services, 'getCategories')
ServiceSpy.mockImplementation(getCategoriesMock)

describe('Header Test', () => {
  it('should render correctly', async () => {
    await act(() => render(<Header />))
    expect(screen.getByTestId('header-wrapper-container')).toMatchSnapshot()
  })
})
