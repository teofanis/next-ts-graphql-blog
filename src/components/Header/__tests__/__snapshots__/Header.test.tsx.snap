// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test should render correctly 1`] = `
<div
  class="container mx-auto px-10 mb-8"
  data-testid="header-wrapper-container"
>
  <div
    class="border-b w-full inline-block dark:border-gray-600 border-blue-400 py-8"
  >
    <div
      class="items-center md:float-left flex justify-between"
    >
      <span
        class="cursor-pointer font-bold text-4xl text-white"
        href="/"
      >
        GraphCMS
      </span>
      <div>
        <button
          aria-label="Toggle Dark Mode"
          class="p-3 h-12 w-12"
          type="button"
        >
          <svg
            class="h-6 w-6 text-white"
            fill="currentColor"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="hidden md:float-left md:contents"
    >
      <span
        class="md:float-right mt-2 align-middle text-white ml-4 font-semibold cursor-pointer"
        href="/category/react"
      >
        React
      </span>
      <span
        class="md:float-right mt-2 align-middle text-white ml-4 font-semibold cursor-pointer"
        href="/category/redux"
      >
        Redux
      </span>
      <span
        class="md:float-right mt-2 align-middle text-white ml-4 font-semibold cursor-pointer"
        href="/category/react-native"
      >
        React Native
      </span>
    </div>
  </div>
</div>
`;
